aptos-move/aptos-vm/src/verifier/transaction_arg_validation.rs

```diff
@@ -331,12 +332,20 @@ fn validate_and_construct<S: MoveResolverExt>(
         )?;
         args.push(arg);
     }
+    let constructor_error = |e: VMError| {
+        if allowed_structs.contains_key("0x1::object::Object") {
```

Comment:
I think we should also pass this through for strings once this is enabled. I think you could just say

```
if are_struct_constructors_enabled { e.into_vm_status() } else { <deserialize-error> }
```

Reply:
I do that. I don't pass the boolean along, so I use the fact that if the allowed_structs contain "Object" then the feature is enabled.

Reply:
So the code as is, gives the behavior you want